internal/provider/resource_packer_channel_test.go

```diff
@@ -0,0 +1,186 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccPackerChannel(t *testing.T) {
+	resourceName := "hcp_packer_channel.production"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t, map[string]bool{"aws": false, "azure": false}) },
+		ProviderFactories: providerFactories,
+		CheckDestroy: func(*terraform.State) error {
+			deleteBucket(t, acctestAlpineBucket, false)
+			return nil
+		},
+
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() { upsertBucket(t, acctestAlpineBucket) },
+				Config:    testConfig(TestAccPackerChannelBasic(acctestAlpineBucket, acctestProductionChannel)),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "author_id"),
+					resource.TestCheckResourceAttr(resourceName, "bucket_name", acctestAlpineBucket),
+					resource.TestCheckResourceAttrSet(resourceName, "created_at"),
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttr(resourceName, "name", acctestProductionChannel),
+					resource.TestCheckResourceAttrSet(resourceName, "organization_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+				),
+			},
+			// Testing that we can import bucket channel created in the previous step and that the
+			// resource terraform state will be exactly the same
+			{
+				ResourceName: resourceName,
+				ImportState:  true,
+				ImportStateIdFunc: func(s *terraform.State) (string, error) {
+					rs, ok := s.RootModule().Resources[resourceName]
+					if !ok {
+						return "", fmt.Errorf("not found: %s", resourceName)
+					}
+
+					bucketName := rs.Primary.Attributes["bucket_name"]
+					channelName := rs.Primary.Attributes["name"]
+					return fmt.Sprintf("%s:%s", bucketName, channelName), nil
+				},
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccPackerChannel_AssignedIteration(t *testing.T) {
+	resourceName := "hcp_packer_channel.production"
+	acctestChannelName := "assinged"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t, map[string]bool{"aws": false, "azure": false}) },
+		ProviderFactories: providerFactories,
+		CheckDestroy: func(*terraform.State) error {
+			deleteBucket(t, acctestAlpineBucket, false)
+			return nil
+		},
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					fingerprint := "channel-assigned-iteration"
+					upsertBucket(t, acctestAlpineBucket)
+					upsertIteration(t, acctestAlpineBucket, fingerprint)
+					itID, err := getIterationIDFromFingerPrint(t, acctestAlpineBucket, fingerprint)
+					if err != nil {
+						t.Fatal(err.Error())
+					}
+					upsertBuild(t, acctestAlpineBucket, fingerprint, itID)
+				},
+				Config: testConfig(testAccPackerChannelAssignedLatestIteration(acctestAlpineBucket, acctestChannelName)),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "iteration_assignment.0.id"),
```

Comment:
What's this `0`?

Reply:
This is a block is a list type so to reference the attributes for the first element in the list, which is the only element, we have to index the results to get the values - https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-types#typelist